test/e2e/framework/pod/utils.go

```diff
@@ -123,15 +133,25 @@ func GetLinuxLabel() *v1.SELinuxOptions {
 // DefaultNonRootUser is the default user ID used for running restricted (non-root) containers.
 const DefaultNonRootUser = 1000
 
+// DefaultNonRootUserName is the default username in Windows used for running restricted (non-root) containers
+const DefaultNonRootUserName = "ContainerUser"
+
 // GetRestrictedPodSecurityContext returns a restricted pod security context.
 // This includes setting RunAsUser for convenience, to pass the RunAsNonRoot check.
 // Tests that require a specific user ID should override this.
 func GetRestrictedPodSecurityContext() *v1.PodSecurityContext {
-	return &v1.PodSecurityContext{
+	psc := &v1.PodSecurityContext{
 		RunAsNonRoot:   pointer.BoolPtr(true),
-		RunAsUser:      pointer.Int64(DefaultNonRootUser),
+		RunAsUser:      GetDefaultNonRootUser(),
 		SeccompProfile: &v1.SeccompProfile{Type: v1.SeccompProfileTypeRuntimeDefault},
 	}
+
+	if NodeOSDistroIs("windows") {
+		psc.WindowsOptions = &v1.WindowsSecurityContextOptions{}
```

Comment:
I think it is better to check if WindowsOptions already exists and then set the fields accordingly.

Reply:
I thought in this case we've created them from scratch so we know it doesn't.